src/interpreted/parser.py

```diff
@@ -383,7 +440,16 @@ def parse_expressions(self) -> list[Expression]:
 
     def parse_expression(self) -> Expression:
         # TODO: extraneous parens can be parsed here.
-        return self.parse_or()
+        return self.parse_as()
+    
+    def parse_as(self) -> Expression:
```

Comment:
You can't put `as` here, because `as` can't be used in any regular expression. It only appears inside an import statement. Finding `as` keyword and taking action should be handled in the code for `parse_import` itself.